sdk/dapp-kit/src/hooks/networkConfig.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { SuiClientOptions } from '@mysten/sui.js/client';
+
+import { useSuiClientContext } from './useSuiClient.js';
+
+export type NetworkConfig<T extends object = object> = SuiClientOptions & {
+	variables?: T;
+};
+
+export function createNetworkConfigs<
+	const T extends Record<string, Config>,
+	Config extends NetworkConfig<Variables> = T[keyof T],
+	Variables extends object = NonNullable<Config['variables']>,
+>(networkConfigs: T) {
+	return {
```

Comment:
I hate depending on function hoisting, this hurts my brain

Reply:
I love function hoisting... it makes things so much easier for me to read, but I can change it if you want.

with this setup its:

1. Inputs
2. Return types
3. the implementation which you probably don't care about

Reply:
The implementation is the _only_ thing I care about.

Reply:
More seriously, I just think people these days don't understand it, especially with `var` being dead and nothing else getting hoisted.